packages/lib/server/queries/teams/index.ts

```diff
@@ -260,6 +260,104 @@ export async function getTeamWithMembers(args: {
   };
 }
 
+export async function getMinimalTeam(args: {
+  id?: number;
+  slug?: string;
+  userId?: number;
+  orgSlug?: string | null;
+  /**
+   * If true, means that you are fetching an organization and not a team
+   */
+  isOrgView?: boolean;
+}) {
+  const { id, slug, userId, orgSlug, isOrgView } = args;
+
+  let lookupBy;
+
+  if (id) {
+    lookupBy = { id, havingMemberWithId: userId };
+  } else if (slug) {
+    lookupBy = { slug, havingMemberWithId: userId };
+  } else {
+    throw new Error("Must provide either id or slug");
+  }
+
+  const arg = {
+    lookupBy,
+    forOrgWithSlug: orgSlug ?? null,
+    isOrg: !!isOrgView,
+    teamSelect: {
+      id: true,
+      name: true,
+      slug: true,
+      isOrganization: true,
+      logoUrl: true,
+      bio: true,
+      hideBranding: true,
+      hideBookATeamMember: true,
+      isPrivate: true,
+      metadata: true,
+      parent: {
+        select: {
+          id: true,
+          slug: true,
+          name: true,
+          isPrivate: true,
+          isOrganization: true,
+          logoUrl: true,
+          metadata: true,
+        },
+      },
+      parentId: true,
+      children: {
+        select: {
+          name: true,
+          slug: true,
+        },
+      },
+      theme: true,
+      brandColor: true,
+      darkBrandColor: true,
+      inviteTokens: {
+        select: {
+          token: true,
+          expires: true,
+          expiresInDays: true,
+          identifier: true,
+        },
+      },
+    },
+  } as const;
+
+  const teamOrOrg = isOrgView ? await getOrg(arg) : await getTeam(arg);
+
+  if (!teamOrOrg) return null;
+
+  // Don't leak invite tokens to the frontend
+  const { inviteTokens, ...teamWithoutInviteTokens } = teamOrOrg;
+
+  // Don't leak stripe payment ids
+  const teamMetadata = teamOrOrg.metadata;
+  const {
+    paymentId: _,
+    subscriptionId: __,
+    subscriptionItemId: ___,
+    ...restTeamMetadata
+  } = teamMetadata || {};
```

Comment:
This is nice but isn't it better to just query what we need instead of having to be aware of all the data we need to cleanup?

Reply:
Metadata type is `Json?` so we cannot skip fields inside metadata.